.circleci/config.yml

```diff
@@ -0,0 +1,52 @@
+version: 2.1
+
+executors:
+  php81:
+    docker:
+      - image: cimg/php:8.1
+    resource_class: small
+  php82:
+    docker:
+      - image: cimg/php:8.2
+    resource_class: small
+
+jobs:
+  analyse:
+    executor: php82
+    steps:
+      - checkout
+      - run: composer update
+      - run: composer analyse
+  coverage:
+    executor: php82
+    steps:
+      - run: sudo pecl install pcov
+      - checkout
+      - run: composer update
+      - run: composer coverage
+  lint:
+    executor: php82
+    steps:
+      - checkout
+      - run: composer update
+      - run: composer lint
+  test:
+    parameters:
+      php_executor:
+        type: string
+    executor: << parameters.php_executor >>
+    steps:
+      - checkout
+      - run: composer update
```

Comment:
If `update` installs newer packages (e.g. instead of just `install`) do these get saved against `composer.lock` at some point?

Reply:
There is no `composer.lock` file

Reply:
Right, right. Very good.